partner_multi_company/models/res_users.py

```diff
@@ -12,16 +12,28 @@ def create(self, vals):
         res = super(ResUsers, self).create(vals)
         if "company_ids" in vals:
             res.partner_id.company_ids = vals["company_ids"]
-        if "company_id" in vals and res.partner_id.company_ids:
-            res.partner_id.company_id = vals["company_id"]
         return res
 
     def write(self, vals):
-        res = super(ResUsers, self).write(vals)
+        res = super(ResUsers, self.with_context(from_res_users=True)).write(vals)
         if "company_ids" in vals:
             for user in self.sudo():
-                if user.partner_id.company_ids:
-                    user.partner_id.company_ids = vals["company_ids"]
+                new_company_ids = []
```

Comment:
Please explain this change of behavior.

Reply:
The idea here is that you add allowed companies to an user, the underlying related partner gets updated if needed.

Before this you can have a user with companies A,B and the related partner with companies A,B and then add company C to the user. This generates an inconsistency as related partner should also have company C, otherwise multicompany rules will raise in different siutations.

Reply:
If the partner has no companies, this does nothing. If partner has no companies, nothing will be assigned.